pennylane/gradients/spsa_gradient.py

```diff
@@ -328,7 +328,7 @@ def _single_shot_batch_result(results):
                     qml.math.tensordot(qml.math.convert_like(_coeffs, res), res, axes=[[0], [0]])
                     + grads
                 )
-            grads = grads / num_directions
+            grads = grads * (1 / num_directions)
```

Comment:
was there a reason you made these division changes?

Reply:
Yes. The (now included) tests using a `SpecialObject` as returned type do not have a division method with `float`s implemented. Of course we could just change the test to include it, but as `finite_diff` and `param_shift` do not require a division method, this change ensure feature parity (although in a very nichy detail, admittedly).